### PR TITLE
add unit test for pixel projection error

### PR DESCRIPTION
**Pull request by edmond-zhu**
add unit test for pixel projection error

### CODE REVIEW

**Comment by edmond-zhu** [comment]
# For pixel projection error unit test:

# Pr for add new unit test for pixel projection error
## license: check if license is ok
## Comments
## warnings in Local build
1. mac: no new warning, only warning: -pie being ignored. It is only used when linking a main executable
2. linux: only job, 
3. windows:  
```
warning LNK4221: This object file does not define any previously undefined public symbols, so it will not be used
by any link operation that consumes this library [F:\work\SVT-AV1\Build\windows\Source\Lib\Common\ASM_SSE
2\COMMON_ASM_SSE2.vcxproj]
```
## warnings on remote ci servers
1. jenkins:
a. clang-format
b. 
2. travis
  a. mac: `-pie being ignored. It is only used when linking a main executable`, can be ignored.
  b. linux: SpaticalFullDistoration wanring, could be fixed by merge the latest code;
3. appvedor
  a. no warnings

## dry run output
~~~
Running main() from C:\Works\AV1\SVT-AV1\third_party\googletest\src\gtest_main.cc
Note: Google Test filter = RST/*.*
[==========] Running 6 tests from 2 test cases.
[----------] Global test environment set-up.
[----------] 3 tests from RST/PixelProjErrorLbdTest
[ RUN      ] RST/PixelProjErrorLbdTest.run_check_output/0
[       OK ] RST/PixelProjErrorLbdTest.run_check_output/0 (4181 ms)
[ RUN      ] RST/PixelProjErrorLbdTest.run_random_size_check_output/0
[       OK ] RST/PixelProjErrorLbdTest.run_random_size_check_output/0 (4139 ms)
[ RUN      ] RST/PixelProjErrorLbdTest.run_extreme_check_output/0
[       OK ] RST/PixelProjErrorLbdTest.run_extreme_check_output/0 (422 ms)
[----------] 3 tests from RST/PixelProjErrorLbdTest (8744 ms total)

[----------] 3 tests from RST/PixelProjErrorHbdTest
[ RUN      ] RST/PixelProjErrorHbdTest.run_check_output/0
[       OK ] RST/PixelProjErrorHbdTest.run_check_output/0 (4131 ms)
[ RUN      ] RST/PixelProjErrorHbdTest.run_random_size_check_output/0
[       OK ] RST/PixelProjErrorHbdTest.run_random_size_check_output/0 (4137 ms)
[ RUN      ] RST/PixelProjErrorHbdTest.run_extreme_check_output/0
[       OK ] RST/PixelProjErrorHbdTest.run_extreme_check_output/0 (427 ms)
[----------] 3 tests from RST/PixelProjErrorHbdTest (8698 ms total)

[----------] Global test environment tear-down
[==========] 6 tests from 2 test cases ran. (17449 ms total)
[  PASSED  ] 6 tests.
~~~

File: test/pixel_proj_err_test.cc [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+#define MAX_DATA_BLOCK 384
+
+namespace {
+using std::make_tuple;
+using svt_av1_test_tool::SVTRandom;
+
+static const int min_test_times = 10;
+
+typedef int64_t (*pixel_proj_fcn)(const uint8_t* src8, int32_t width,
```
Comment (Cidana-Developers) [inline test/pixel_proj_err_test.cc]: Type should be Camel_case, like PixelProjFunc

Comment (edmond-zhu) [reply test/pixel_proj_err_test.cc]: done

**Comment by Cidana-Developers** [comment]
Merged.

**Pull request closed without merging**